src/Silverback.Integration.Kafka/Messaging/Messages/KafkaKeyHelper.cs

```diff
@@ -2,43 +2,86 @@
 // This code is licensed under MIT license (see LICENSE file for details)
 
 using System;
-using System.Globalization;
+using System.Collections.Generic;
 using System.Linq;
-using Silverback.Messaging.HealthChecks;
+using System.Reflection;
+using System.Text;
 
 namespace Silverback.Messaging.Messages
 {
     internal static class KafkaKeyHelper
     {
+        private static readonly Dictionary<Type, PropertyInfo[]> _typeCache = new();
```

Comment:
I'm surprised I didn't have this already. This is very old code but still...

Thanks!